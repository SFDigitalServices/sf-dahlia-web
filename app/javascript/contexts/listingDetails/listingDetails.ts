interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
