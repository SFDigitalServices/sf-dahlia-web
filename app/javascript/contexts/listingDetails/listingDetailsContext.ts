import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
