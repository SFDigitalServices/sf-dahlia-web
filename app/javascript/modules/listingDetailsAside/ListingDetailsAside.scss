.fcfs-aside {
    --card-border-radius: rounded-none;
    --card-background-color: var(--bloom-color-gray-200);
    --card-footer-background-color: var(--bloom-color-gray-200);
    --card-border-width: border-b-1;

    .aside-card-header {
        font-family: var(--seeds-font-alt-sans);
        font-size: var(--seeds-font-size-lg);
        font-weight: var(-seeds-font-weight-semibold);
        line-height: var(--seeds-line-height-6);
    }

    .aside-card-section {
        font-family: var(--seeds-font-sans);
        font-size: var(--seeds-font-size-sm);
        font-weight: var(--seeds-font-weight-regular);
        line-height: var(--seeds-line-height-5);
    }
}
