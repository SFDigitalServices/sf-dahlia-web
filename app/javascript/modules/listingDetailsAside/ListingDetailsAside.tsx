import React from "react"
import { ListingDetailItem, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsInfoSession } from "./ListingDetailsInfoSession"
import { ListingDetailsProcess } from "./ListingDetailsProcess"
import { isOpen } from "../../util/listingUtil"
import { ListingDetailsApply } from "./ListingDetailsApply"
import { ListingDetailsApplicationDate } from "./ListingDetailsApplicationDate"
import { ListingDetailsLotteryResults } from "../listingDetailsLottery/ListingDetailsLotteryResults"

export interface ListingDetailsSidebarProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsAside = ({ listing, imageSrc }: ListingDetailsSidebarProps) => {
  const isApplicationOpen = isOpen(listing)

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.process.header")}
      subtitle={t("listings.process.subheader")}
      hideHeader={true}
      desktopClass="header-hidden"
    >
      <aside className="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white">
        <div className="hidden md:block">
          <ListingDetailsApplicationDate isApplicationOpen={isApplicationOpen} listing={listing} />
          <ListingDetailsLotteryResults listing={listing} />
          {isApplicationOpen && <ListingDetailsInfoSession listing={listing} />}
          <ListingDetailsApply listing={listing} />
          <ListingDetailsProcess listing={listing} />
        </div>
      </aside>
    </ListingDetailItem>
  )
}
