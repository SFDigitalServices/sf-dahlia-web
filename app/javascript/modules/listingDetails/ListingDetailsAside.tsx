import React from "react"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsInfoSession } from "../listingDetailsAside/ListingDetailsInfoSession"
import { ListingDetailsApply } from "../listingDetailsAside/ListingDetailsApply"
import { ListingDetailsProcess } from "../listingDetailsAside/ListingDetailsProcess"
import { isOpen } from "../../util/listingUtil"
import { ListingDetailsApplicationDate } from "../listingDetailsAside/ListingDetailsApplicationDate"

export interface ListingDetailsSidebarProps {
  listing: RailsListing
}

export const ListingDetailsAside = ({ listing }: ListingDetailsSidebarProps) => {
  const isApplicationOpen = isOpen(listing)

  return (
    <aside className="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white">
      <div className="hidden md:block">
        <ListingDetailsApplicationDate listing={listing} />
        {isApplicationOpen && <ListingDetailsInfoSession listing={listing} />}
        <ListingDetailsApply listing={listing} />
        <ListingDetailsProcess listing={listing} />
      </div>
    </aside>
  )
}
