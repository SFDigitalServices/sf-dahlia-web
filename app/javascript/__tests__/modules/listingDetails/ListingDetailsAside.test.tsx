import React from "react"
import renderer from "react-test-renderer"
import { ListingDetailsAside } from "../../../modules/listingDetails/ListingDetailsAside"

describe("Listing Details Aside", () => {
  it("displays Application Deadline when due date has not passed", () => {
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    const tree = renderer
      .create(
        <ListingDetailsAside
          listing={{ Application_Due_Date: "2050-01-01T01:00:00.000+0000" } as any}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
  it("displays Applications Closed when due date has passed", () => {
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    const tree = renderer
      .create(
        <ListingDetailsAside
          listing={{ Application_Due_Date: "2000-01-01T01:00:00.000+0000" } as any}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
