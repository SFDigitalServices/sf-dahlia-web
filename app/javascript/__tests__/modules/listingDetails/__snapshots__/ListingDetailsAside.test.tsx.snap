// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsAside renders ListingDetailsAside component fcfs 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="responsive-content-item header-hidden"
    >
      <header
        class="detail-header md:hidden"
      >
        <span
          class="detail-header__image-container"
        >
          <img
            alt=""
            class="detail-header__image "
            src=""
          />
        </span>
        <hgroup
          class="detail-header__hgroup"
        >
          <h2
            class="detail-header__title"
          >
            Process
          </h2>
          <span
            class="detail-header__subtitle"
          >
            Important dates and contact information
          </span>
          <span
            class="ui-icon ui-medium"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 32 32"
            >
              <path
                d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
              />
            </svg>
          </span>
        </hgroup>
      </header>
      <aside
        class="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white"
      >
        <div
          class="hidden md:block"
        >
          <div
            class="w-full mb-8 md:mb-0"
          >
            <div
              class="seeds-common-message is-fullwidth seeds-message justify-start leading-5"
              data-variant="primary"
            >
              <span
                class="ui-icon ui-medium"
              >
                <svg
                  fill="currentColor"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                  />
                  <path
                    d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                  />
                </svg>
              </span>
              <span
                data-part="content"
              >
                Applications open: January 1, 2050 at 1:00Â AM Pacific Time
              </span>
            </div>
          </div>
          <article
            class="seeds-card fcfs-no-lottery"
            data-spacing="base"
          >
            <header
              class="seeds-card-header font-bold fcfs-no-lottery-header"
            >
              No lottery required
            </header>
            <div
              class="seeds-card-section fcfs-no-lottery-section"
            >
              Your application will be considered in the order it comes in - first come, first served.
            </div>
            <footer
              class="seeds-card-footer ml-6 mb-6 underline fcfs-no-lottery-section"
            >
              <a
                class="seeds-link"
                href="https://www.sf.gov/step-by-step/buy-home-without-entering-lottery"
              >
                Learn more about how it works
              </a>
            </footer>
          </article>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block"
            >
              <h4
                class="text__underline-weighted"
              >
                Information Sessions
              </h4>
              <div
                class="false"
              >
                <p
                  class="text text-gray-800 pb-2 flex justify-between items-center"
                >
                  <span
                    class="inline-block text-sm uppercase normal-case"
                  >
                    November 19, 2019
                  </span>
                  <span
                    class="inline-block text-xs font-bold ml-5 font-alt-sans"
                  >
                    6:00 PM
                  </span>
                </p>
                <div
                  class="text-sm text-gray-700 false"
                >
                  <div
                    class="flex flex-col"
                  >
                    <span
                      class="links-space translate"
                    >
                      <span>
                        MOHCD
                      </span>
                    </span>
                    <span>
                      1 South Van Ness Ave, Fl 5, San Francisco
                    </span>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="md:px-0 px-2"
          >
            <section
              class="aside-block fcfs-bmr-how-to-apply"
            >
              <h4
                class="text__underline-weighted"
              >
                How to Apply
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <div
                  class="fcfs-bmr-how-to-apply__list"
                >
                  <ol
                    class="numbered-list text-black text-base"
                  >
                    <li>
                      Fill out PDF application
                    </li>
                    <li>
                      Gather documents
                    </li>
                    <li>
                      When you are ready, submit application
                    </li>
                  </ol>
                </div>
                <a
                  class="button is-primary w-full transition"
                  href="/listings/a0W0P00000GlKfBUAV/how-to-apply"
                  tabindex="0"
                >
                  Learn more and apply
                </a>
              </div>
            </section>
          </div>
          <section
            class="aside-block see-the-unit"
          >
            <div
              class="see-the-unit__heading"
            >
              <h4
                class="text__underline-weighted"
              >
                See the Unit
              </h4>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                Open houses
              </h1>
              <div>
                <div
                  class="open-houses"
                >
                  <section
                    class=""
                  >
                    <div
                      class="pb-3"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 3, 2019
                        </span>
                        <span
                          class="inline-block text-xs font-bold ml-5 font-alt-sans"
                        >
                          2:00 PM - 6:00 PM
                        </span>
                      </p>
                    </div>
                    <div
                      class="pb-3"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 7, 2019
                        </span>
                        <span
                          class="inline-block text-xs font-bold ml-5 font-alt-sans"
                        >
                          10:00 AM - 12:00 PM
                        </span>
                      </p>
                      <div
                        class="text-sm text-gray-700 pb-3"
                      >
                        <div
                          class="flex flex-col"
                        >
                          <span
                            class="links-space translate"
                          >
                            <span>
                              SFPL
                            </span>
                          </span>
                          <span>
                            1 Library Way, San Francisco
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="false"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 8, 2019
                        </span>
                      </p>
                    </div>
                  </section>
                </div>
              </div>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                See details online
              </h1>
              <div>
                <div
                  class="flex-row space-y-1"
                >
                  <div>
                    <a
                      class="seeds-link no-underline"
                      href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
                    >
                      See the unit on MLS
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                Make an appointment
              </h1>
              <div>
                <p
                  class="text-sm pb-3"
                >
                  Call, email, or text to request a tour.
                </p>
                <p>
                  Matthew Burns
                </p>
                <p
                  class="text-gray-700 text-sm"
                >
                  Leasing Agent
                </p>
                <div
                  class="pt-2"
                >
                  <p
                    class="pb-1"
                  >
                    <a
                      href="tel:415 5552323"
                    >
                      <span
                        class="ui-icon ui-medium pr-2"
                      >
                        <svg
                          fill="#0077DA"
                          viewBox="0 0 32 32"
                        >
                          <path
                            d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                          />
                        </svg>
                      </span>
                      Call (415) 555-2323
                    </a>
                  </p>
                  <p
                    class="pb-3"
                  >
                    <a
                      href="mailto:ae@email.com"
                    >
                      <span
                        class="pr-2"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-envelope "
                          data-icon="envelope"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                      Email
                    </a>
                  </p>
                </div>
                <h1
                  class="text-heading-sm pb-1"
                >
                  Office hours
                </h1>
                <p
                  class="text-sm"
                >
                  8-4 Mon-Thurs
9-4 Fri
10-2 Weekends
                </p>
              </div>
            </div>
          </section>
          <section
            class="aside-block "
          >
            <h4
              class="text__underline-weighted"
            >
              Need Help?
            </h4>
            <div
              class="text-sm text-gray-750"
            >
              <a
                class="button w-full transition"
                href="https://www.homeownershipsf.org/buyerapplications/"
                target="_blank"
              >
                Visit HomeownershipSF
              </a>
            </div>
          </section>
          <section
            class="aside-block "
          >
            <h4
              class="text__underline-weighted"
            >
              Expected move-in date
            </h4>
            <div
              class="text-sm text-gray-750"
            >
              June 2020
            </div>
          </section>
          <div
            class="border-b border-gray-400 md:border-b-0 last:border-b-0"
          >
            <section
              class="aside-block "
            >
              <h4
                class="text__underline-weighted"
              >
                Housing Program
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <a
                  class="text-base"
                  href="https://sfmohcd.org/for-buyers"
                  target="_blank"
                >
                  Below Market Rate
                </a>
              </div>
            </section>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0 last:border-b-0"
          >
            <section
              class="aside-block "
            >
              <div
                class="text-sm text-gray-750"
              >
                <p>
                  Listing Updated: April 11, 2022
                </p>
              </div>
            </section>
          </div>
        </div>
      </aside>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`ListingDetailsAside renders ListingDetailsAside component rental 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="responsive-content-item header-hidden"
    >
      <header
        class="detail-header md:hidden"
      >
        <span
          class="detail-header__image-container"
        >
          <img
            alt=""
            class="detail-header__image "
            src=""
          />
        </span>
        <hgroup
          class="detail-header__hgroup"
        >
          <h2
            class="detail-header__title"
          >
            Process
          </h2>
          <span
            class="detail-header__subtitle"
          >
            Important dates and contact information
          </span>
          <span
            class="ui-icon ui-medium"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 32 32"
            >
              <path
                d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
              />
            </svg>
          </span>
        </hgroup>
      </header>
      <aside
        class="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white"
      >
        <div
          class="hidden md:block"
        >
          <div
            class="w-full mb-8 md:mb-0"
          >
            <div
              class="w-full mb-8 md:mb-0"
            >
              <div
                class="application-status text-xs text-gray-800 bg-alert-light place-content-center"
              >
                <span
                  class="ui-icon ui-medium"
                >
                  <svg
                    fill="var(--bloom-color-red-700)"
                    viewBox="0 0 32 32"
                  >
                    <path
                      d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                    />
                    <path
                      d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                    />
                  </svg>
                </span>
                <span>
                  Applications Closed Jan 1, 2000 at 1:00 AM
                </span>
              </div>
            </div>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block "
            >
              <h4
                class="text__underline-weighted"
              >
                Lottery
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <p
                  class="flex justify-between mb-4"
                >
                  <span>
                    March 22, 2023
                  </span>
                  <span
                    class="font-bold"
                  >
                    6:00 pm
                  </span>
                </p>
                <div
                  class="text-gray-700"
                >
                  <div>
                    <p>
                      The lottery will not be in person, but instead will be a virtual lottery.
                    </p>
                    <p>
                      Â 
                    </p>
                    <p>
                      Come back to this page on the day of the lottery for a link you can click on to watch the lottery live.
                    </p>
                    <p>
                      <br />
                    </p>
                    <p>
                      <br />
                    </p>
                  </div>
                </div>
                <p
                  class="mt-4 text-gray-700"
                >
                  An open lottery will be held to draw numbers to determine selection. Attendance is optional.
                </p>
              </div>
            </section>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block "
            >
              <h4
                class="text__underline-weighted"
              >
                Lottery Results
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <p
                  class="mb-4"
                >
                  March 23, 2023
                </p>
                <p
                  class="text-gray-700"
                >
                  Complete lottery results will be posted on this date by 5 pm.
                </p>
              </div>
            </section>
          </div>
          <div
            class="open-houses border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block"
            >
              <h4
                class="text__underline-weighted"
              >
                Open Houses
              </h4>
              <div
                class="false"
              >
                <p
                  class="text text-gray-800 pb-2 flex justify-between items-center"
                >
                  <span
                    class="inline-block text-sm uppercase normal-case"
                  >
                    February 23, 2018
                  </span>
                  <span
                    class="inline-block text-xs font-bold ml-5 font-alt-sans"
                  >
                    9:00AM - 12:00PM
                  </span>
                </p>
                <div
                  class="text-sm text-gray-700 false"
                >
                  <div
                    class="flex flex-col"
                  >
                    <span
                      class="links-space translate"
                    >
                      <span>
                        Library
                      </span>
                    </span>
                    <span>
                      123 Van ness, San Francisco
                    </span>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block"
            >
              <h4
                class="text__underline-weighted"
              >
                What to Expect
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <span>
                  Applicants will be contacted by the agent in lottery rank order until vacancies are filled.
                </span>
                <div
                  class="mt-2"
                >
                  <div>
                    <button
                      aria-expanded="false"
                      aria-label="Show more about what to expect"
                      class="button is-unstyled m-0 no-underline has-toggle"
                      type="button"
                    >
                      Show more
                    </button>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </aside>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`ListingDetailsAside renders ListingDetailsAside component sales 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="responsive-content-item header-hidden"
    >
      <header
        class="detail-header md:hidden"
      >
        <span
          class="detail-header__image-container"
        >
          <img
            alt=""
            class="detail-header__image "
            src=""
          />
        </span>
        <hgroup
          class="detail-header__hgroup"
        >
          <h2
            class="detail-header__title"
          >
            Process
          </h2>
          <span
            class="detail-header__subtitle"
          >
            Important dates and contact information
          </span>
          <span
            class="ui-icon ui-medium"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 32 32"
            >
              <path
                d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
              />
            </svg>
          </span>
        </hgroup>
      </header>
      <aside
        class="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white"
      >
        <div
          class="hidden md:block"
        >
          <div
            class="w-full mb-8 md:mb-0"
          >
            <div
              class="w-full mb-8 md:mb-0"
            >
              <div
                class="application-status text-xs text-gray-800 bg-primary-light place-content-center"
              >
                <span
                  class="ui-icon ui-medium"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 32 32"
                  >
                    <path
                      d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                    />
                    <path
                      d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                    />
                  </svg>
                </span>
                <span>
                  Application Deadline Jan 1, 2050 at 1:00 AM
                </span>
              </div>
            </div>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block"
            >
              <h4
                class="text__underline-weighted"
              >
                Information Sessions
              </h4>
              <div
                class="false"
              >
                <p
                  class="text text-gray-800 pb-2 flex justify-between items-center"
                >
                  <span
                    class="inline-block text-sm uppercase normal-case"
                  >
                    November 19, 2019
                  </span>
                  <span
                    class="inline-block text-xs font-bold ml-5 font-alt-sans"
                  >
                    6:00 PM
                  </span>
                </p>
                <div
                  class="text-sm text-gray-700 false"
                >
                  <div
                    class="flex flex-col"
                  >
                    <span
                      class="links-space translate"
                    >
                      <span>
                        MOHCD
                      </span>
                    </span>
                    <span>
                      1 South Van Ness Ave, Fl 5, San Francisco
                    </span>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="md:px-0 px-2"
          >
            <section
              class="aside-block "
            >
              <h4
                class="text__underline-weighted"
              >
                How to Apply
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <p
                  class="mb-4"
                >
                  <span>
                    Make sure you fulfill the 
                    <a
                      href="https://sf.gov/determine-if-you-can-buy-affordable-housing-program"
                      target="_blank"
                    >
                      eligibility requirements
                    </a>
                     for this listing before you begin.
                  </span>
                </p>
                <p
                  class="mb-4"
                >
                  Your loan pre-approval letter must be dated within the past 120 days, and your Homebuyer Education verification must be dated within the past year.
                </p>
                <a
                  class="button is-primary w-full transition"
                  href="listings/a0W0P00000GlKfBUAV/apply-welcome/intro"
                  tabindex="0"
                >
                  Apply Online
                </a>
              </div>
            </section>
          </div>
          <section
            class="aside-block see-the-unit"
          >
            <div
              class="see-the-unit__heading"
            >
              <h4
                class="text__underline-weighted"
              >
                See the Unit
              </h4>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                Open houses
              </h1>
              <div>
                <div
                  class="open-houses"
                >
                  <section
                    class=""
                  >
                    <div
                      class="pb-3"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 3, 2019
                        </span>
                        <span
                          class="inline-block text-xs font-bold ml-5 font-alt-sans"
                        >
                          2:00 PM - 6:00 PM
                        </span>
                      </p>
                    </div>
                    <div
                      class="pb-3"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 7, 2019
                        </span>
                        <span
                          class="inline-block text-xs font-bold ml-5 font-alt-sans"
                        >
                          10:00 AM - 12:00 PM
                        </span>
                      </p>
                      <div
                        class="text-sm text-gray-700 pb-3"
                      >
                        <div
                          class="flex flex-col"
                        >
                          <span
                            class="links-space translate"
                          >
                            <span>
                              SFPL
                            </span>
                          </span>
                          <span>
                            1 Library Way, San Francisco
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="false"
                    >
                      <p
                        class="text text-gray-800 pb-2 flex justify-between items-center"
                      >
                        <span
                          class="inline-block text-sm uppercase normal-case"
                        >
                          December 8, 2019
                        </span>
                      </p>
                    </div>
                  </section>
                </div>
              </div>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                See details online
              </h1>
              <div>
                <div
                  class="flex-row space-y-1"
                >
                  <div>
                    <a
                      class="seeds-link no-underline"
                      href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
                    >
                      See the unit on MLS
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="see-the-unit__subsection"
            >
              <h1
                class="text-heading-md pb-3"
              >
                Make an appointment
              </h1>
              <div>
                <p
                  class="text-sm pb-3"
                >
                  Call, email, or text to request a tour.
                </p>
                <p>
                  Matthew Burns
                </p>
                <p
                  class="text-gray-700 text-sm"
                >
                  Leasing Agent
                </p>
                <div
                  class="pt-2"
                >
                  <p
                    class="pb-1"
                  >
                    <a
                      href="tel:415 5552323"
                    >
                      <span
                        class="ui-icon ui-medium pr-2"
                      >
                        <svg
                          fill="#0077DA"
                          viewBox="0 0 32 32"
                        >
                          <path
                            d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                          />
                        </svg>
                      </span>
                      Call (415) 555-2323
                    </a>
                  </p>
                  <p
                    class="pb-3"
                  >
                    <a
                      href="mailto:ae@email.com"
                    >
                      <span
                        class="pr-2"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-envelope "
                          data-icon="envelope"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                      Email
                    </a>
                  </p>
                </div>
                <h1
                  class="text-heading-sm pb-1"
                >
                  Office hours
                </h1>
                <p
                  class="text-sm"
                >
                  8-4 Mon-Thurs
9-4 Fri
10-2 Weekends
                </p>
              </div>
            </div>
          </section>
          <section
            class="aside-block "
          >
            <h4
              class="text__underline-weighted"
            >
              Need Help?
            </h4>
            <div
              class="text-sm text-gray-750"
            >
              <a
                class="button w-full transition"
                href="https://www.homeownershipsf.org/buyerapplications/"
                target="_blank"
              >
                Visit HomeownershipSF
              </a>
            </div>
          </section>
          <section
            class="aside-block "
          >
            <h4
              class="text__underline-weighted"
            >
              Expected move-in date
            </h4>
            <div
              class="text-sm text-gray-750"
            >
              June 2020
            </div>
          </section>
          <div
            class="border-b border-gray-400 md:border-b-0"
          >
            <section
              class="aside-block"
            >
              <h4
                class="text__underline-weighted"
              >
                What to Expect
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <span>
                  Applicants will be contacted by the agent in lottery rank order until vacancies are filled.
                </span>
                <div
                  class="mt-2"
                >
                  <div>
                    <button
                      aria-expanded="false"
                      aria-label="Show more about what to expect"
                      class="button is-unstyled m-0 no-underline has-toggle"
                      type="button"
                    >
                      Show more
                    </button>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0 last:border-b-0"
          >
            <section
              class="aside-block "
            >
              <h4
                class="text__underline-weighted"
              >
                Housing Program
              </h4>
              <div
                class="text-sm text-gray-750"
              >
                <a
                  class="text-base"
                  href="https://sfmohcd.org/for-buyers"
                  target="_blank"
                >
                  Below Market Rate
                </a>
              </div>
            </section>
          </div>
          <div
            class="border-b border-gray-400 md:border-b-0 last:border-b-0"
          >
            <section
              class="aside-block "
            >
              <div
                class="text-sm text-gray-750"
              >
                <p>
                  Listing Updated: April 11, 2022
                </p>
              </div>
            </section>
          </div>
        </div>
      </aside>
    </li>
  </ul>
</DocumentFragment>
`;
